docs/tutorials/tutorial.md

```diff
@@ -0,0 +1,60 @@
+
+- Support any existing tooling for building container images in Kubernetes clusters.
+- Minimize the burden of learning a new tool for building images, by abstracting it from users.
+- Simplify the user experience when building images via a standardize minimal API.
+- Allow users to re-use their existing cluster for building and deploying container images.
+
+## Concepts
+
+| Concept     | Description |
+| ----------- | ----------- |
+| **`Strategy`**      | Refers to a particular tool that will be used when building a container image, such as Kaniko, Buildah, ko, etc. |
+| **`Build`**   | Resource used to define a build configuration. |
+| **`BuildRun`**   | Resource used to start the image build mechanism. |
+| **`BuildStrategy/ClusterBuildStrategy`**   | Resource that holds a template that dictates how to build via a particular strategy. |
+| **`Dockerfile-less strategy`**   | Is a category given to strategies that can build container images from source code, without the notion of a Dockerfile. |
+| **`Dockerfile-based strategy`**   | Is a category given to strategies that can build container images from source code, with a reference to a Dockerfile. |
+
+With the above concepts in mind, lets see how they all play together.
+
+## Strategies
+
+Shipwright ships with a set of strategies that are available across the cluster.
+
+The default installation includes these [buildstrategies](/docs/buildstrategies.md):
+
+* [Source-to-Image](docs/buildstrategies.md#source-to-image)
+* [Buildpacks-v3](docs/buildstrategies.md#buildpacks-v3)
+* [Buildah](docs/buildstrategies.md#buildah)
+* [Kaniko](docs/buildstrategies.md#kaniko)
+* [ko](docs/buildstrategies.md#ko)
+
+For more information about strategies see the related [docs](/docs/buildstrategies.md).
+
+## Examples
+
+### [Example with Kaniko](/docs/tutorials/building_with_kaniko.md)
+
+### [Example with Paketo](/docs/tutorials/building_with_paketo.md)
+
+Depending on your source code you might want to try a specific example. The following table serves as a guide to help you understand which
+strategy to choose:
+
+| Sample code | Repository | ContextDir | Strategy Type | Strategy to use |
+| ----------- | ----------- | ------------- | ------------- | ------------- |
+| A go app with a Dockerfile | [shipwright-io/sample-go](https://github.com/shipwright-io/sample-go) | `/docker-build` | Dockerfile-based | Kaniko, Buildah |
+| A go app | [shipwright-io/sample-go](https://github.com/shipwright-io/sample-go) | `/source-build` | Dockerfile-less | Paketo,Heroku |
+| A ruby app | [shipwright-io/sample-ruby](https://github.com/shipwright-io/sample-ruby) | `/source-build` | Dockerfile-less | Paketo,Heroku |
+| A java app with a Dockerfile | [hipwright-io/sample-jave](https://github.com/shipwright-io/sample-java) | `/docker-build` | Dockerfile-based | Kaniko, Buildah |
+| Shipwright/Build | [shipwright-io/build](https://github.com/shipwright-io/build) |  `/cmd/manager` | Dockerfile-less | Paketo,Heroku, Ko |
```

Comment:
Buildpacks cannot be used for this repo. This fails because there is no main package in the root directory. Just verified:

```bash
$ ibmcloud ce build create -n shipwright-build -src https://github.com/shipwright-io/build -rs cr-sascha -image registry.saschaschwarze.de/sascha-codeengine/shipwright-build-controller -strategy buildpacks
Creating build 'shipwright-build'...
OK
$ ibmcloud ce buildrun submit -b shipwright-build
Submitting build run 'shipwright-build-run-210329-121425183'...
Run 'ibmcloud ce buildrun get -n shipwright-build-run-210329-121425183' to check the build run status.
OK
$ build-logs shipwright-build-run-210329-121425183 -f
[..]
Paketo Go Build Buildpack 0.3.0
  Executing build process
    Running 'go build -o /layers/paketo-buildpacks_go-build/targets/bin -buildmode pie .'
      Failed after 216ms
failed to execute 'go build': exit status 1
        package github.com/shipwright-io/build: build constraints exclude all Go files in /workspace/source
ERROR: failed to build: exit status 1
```

Reply:
Remove it from there, I just left ko for now